src/ekat/ekat_workspace_impl.hpp

```diff
@@ -241,6 +260,42 @@ void WorkspaceManager<T, D>::Workspace::take_many_contiguous_unsafe(
   m_team.team_barrier();
 }
 
+template <typename T, typename D>
+template <typename S>
+KOKKOS_INLINE_FUNCTION
+Unmanaged<typename WorkspaceManager<T, D>::template view_1d<S> >
+WorkspaceManager<T, D>::Workspace::take_n_size_block(
+  const char* name, const int n_sub_blocks) const
+{
+#ifndef NDEBUG
+  change_num_used(n_sub_blocks);
+  // Verify contiguous
+  for (int n = 0; n < n_sub_blocks; ++n) {
+    const auto space = m_parent.get_space_in_slot<S>(m_ws_idx, m_next_slot + n);
+    EKAT_KERNEL_ASSERT(m_parent.get_next<S>(space) == m_next_slot + n + 1);
+    m_team.team_barrier();
```

Comment:
@bartgol @jgfouca Feel free to ignore this till next week, but without this team barrier (or if the team barrier exists outside of this for-loop), `m_parent.get_next<S>(space)` has a nonsense value for some of the spaces when run with NUM_THREADS>1. Even when there is just one call to this function (i.e., teampolicy(1) and only taking slots via this method).

Any ideas why this would be the case? It seems almost identical the check in the above function, so I don't know why a barrier is needed here.

Reply:
That is strange. Maybe try removing this barrier and running with valgrind?